modules/persistence/tests/services/pages.test.ts

```diff
@@ -122,5 +143,157 @@ describe('pages module', () => {
       res = await mockDb.collection(collection).find(findQuery).toArray();
       expect(res).toHaveLength(0);
     });
+
+    describe('static assets', () => {
+      const sampleStaticAssets = [
+        { checksum: '1234567890', key: '/images/new-asset.png' },
+        { checksum: '0987654321', key: '/images/another-new-asset.png' },
+      ];
+
+      const createSamplePage = (pagePrefix: string, withAssets?: boolean) => {
+        const samplePage: Page = {
+          page_id: `${pagePrefix}/page0.txt`,
+          filename: 'page0.txt',
+          ast: { foo: 'foo', bar: { foo: 'foo' } },
+          static_assets: [],
+        };
+
+        if (withAssets) {
+          samplePage.static_assets = sampleStaticAssets;
+        }
+
+        return samplePage;
+      };
+
+      it('should return empty assets when there are no assets', async () => {
+        // Setup for empty static assets
+        const pagePrefix = generatePagePrefix();
+        const page = createSamplePage(pagePrefix);
+        await _updatePages([page], collection);
+
+        const findQuery = { page_id: page.page_id };
+        let res = await mockDb.collection<UpdatedPage>(collection).findOne(findQuery);
+        expect(res).toBeTruthy();
+        if (res) {
```

Comment:
makes sense. not ideal but i see the reasoning 👍

Reply:
I ended up reverting the if statements and instead left a comment to disable the non-null assertion warnings for only that file (in case we want to have that warning active elsewhere). I think this helps keep things clean overall, and combined with the `expect(res).toBeTruthy()`, we help ensure that the non-null assertions are safe. Let me know if you have any issues with this!